### PR TITLE
Refactor Event Subscription

### PR DESCRIPTION
**Pull request by alexanderbez**
- Remove usage of flattened events, i.e. `map[string][]string`, from the pubsub/events surface area
- Change `ResultEvent.Events` to `[]abci.Event` instead of `map[string][]string`

closes: #5133

### CODE REVIEW

File: libs/pubsub/query/query_test.go [hunk 1]
```diff
@@ -2,15 +2,38 @@ package query_test
...
 	"time"
 
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-
+	abci "github.com/tendermint/tendermint/abci/types"
 	"github.com/tendermint/tendermint/libs/pubsub/query"
 )
 
+func expandEvents(flattenedEvents map[string][]string) []abci.Event {
```
Comment (alexanderbez) [inline libs/pubsub/query/query_test.go:14]: I added this helper method to avoid having to modify the massive `TestMatches` which would be a huge PIA.

**Review by marbar3778 — COMMENTED** [review]
LGTM. 

cc @colin-axner @AdityaSripal @ethanfrey for visibility

**Review by marbar3778 — APPROVED** [review]

**Comment by marbar3778** [comment]
also closes #5963

**Review by tychoish — APPROVED** [review]

**Pull request merged**